ddl/ddl_api.go

```diff
@@ -768,11 +768,33 @@ func buildColumnsAndConstraints(
 	colMap := make(map[string]*table.Column, len(colDefs))
 
 	for i, colDef := range colDefs {
+		if field_types.TiDBStrictIntegerDisplayWidth {
+			switch colDef.Tp.GetType() {
+			case mysql.TypeTiny:
+				// No warning for BOOL-like tinyint(1)
+				if colDef.Tp.GetFlen() > -1 && colDef.Tp.GetFlen() != 1 {
```

Comment:
Please use `colDef.Tp.GetFlen() != UnspecifiedLength` instead of `> -1` Or are there other cases using negative length other than UnspecifiedLength?